modules/mysql.lua

```diff
@@ -43,22 +52,74 @@ local sql           = l.Cg((l.P(1) - sql_end)^0 * sql_end, "Payload")
+                       * l.Cg(l.Cc"B", "representation")), "Bytes_sent")
+                       * ("  Tmp_tables: " * l.Cg(integer, "Tmp_tables")
+                       * "  Tmp_disk_tables: " * l.Cg(integer, "Tmp_disk_tables")
+                       * "  Tmp_table_sizes: " * l.Cg(l.Ct(l.Cg(integer, "value")
+                       * l.Cg(l.Cc"B", "representation")), "Tmp_table_sizes"))^0 * sep
+
+local stored_routine   = "# Stored routine: " * l.Cg((l.alnum + l.S"_.-")^1, "Stored_routine") * sep
+
+local query_plan_info  = "# QC_Hit: " * l.Cg(yes_no, "QC_hit")
+                       * "  Full_scan: " * l.Cg(yes_no, "Full_scan")
+                       * "  Full_join: " * l.Cg(yes_no, "Full_join")
+                       * "  Tmp_table: " * l.Cg(yes_no, "Tmp_table")
+                       * "  Tmp_table_on_disk: " * l.Cg(yes_no, "Tmp_table_on_disk") * sep
+                       * "# Filesort: " * l.Cg(yes_no, "Filesort")
+                       * "  Filesort_on_disk: " * l.Cg(yes_no, "Filesort_on_disk")
+                       * "  Merge_passes: " * l.Cg(integer, "Merge_passes") * sep
+
+local innodb_usage     = "#   InnoDB_IO_r_ops: " * l.Cg(integer, "InnoDB_IO_r_ops")
+                       * "  InnoDB_IO_r_bytes: " * l.Cg(l.Ct(l.Cg(integer, "value")
+                       * l.Cg(l.Cc"B", "representation")), "InnoDB_IO_r_bytes")
+                       * "  InnoDB_IO_r_wait: " * l.Cg(l.Ct(l.Cg(float, "value")
+                       * l.Cg(l.Cc"s", "representation")), "InnoDB_IO_r_wait") * sep
+                       * "#   InnoDB_rec_lock_wait: " * l.Cg(l.Ct(l.Cg(float, "value")
+                       * l.Cg(l.Cc"s", "representation")), "InnoDB_rec_lock_wait")
+                       * "  InnoDB_queue_wait: " * l.Cg(l.Ct(l.Cg(float, "value")
+                       * l.Cg(l.Cc"s", "representation")), "InnoDB_queue_wait") * sep
+                       * "#   InnoDB_pages_distinct: " * l.Cg(integer, "InnoDB_pages_distinct") * sep
 
 slow_query_grammar          = l.Ct(time^0 * user * l.Cg(l.Ct(query), "Fields") * use_db^0 * set * admin^0 * sql)
 mariadb_slow_query_grammar  = l.Ct(time^0 * user * l.Cg(l.Ct(thread_id * query * full_scan^0), "Fields") * use_db^0 * set * admin^0 * sql)
+percona_slow_query_grammar  = l.Ct(time^0 * percona_user * l.Cg(l.Ct(conn_id^0 * info^0 * percona_query * memory_footprint^0 * stored_routine^0 * query_plan_info^0 * innodb_usage^0), "Fields") * use_db^0 * set * admin^0 * sql)
 
-short_slow_query_grammar        = l.Ct(l.Cg(l.Ct(query), "Fields") * use_db^0 * set * admin^0 * sql)
-mariadb_short_slow_query_grammar= l.Ct(l.Cg(l.Ct(thread_id * query * full_scan^0), "Fields") * use_db^0 * set * admin^0 * sql)
+short_slow_query_grammar         = l.Ct(l.Cg(l.Ct(query), "Fields") * use_db^0 * set * admin^0 * sql)
+mariadb_short_slow_query_grammar = l.Ct(l.Cg(l.Ct(thread_id * query * full_scan^0), "Fields") * use_db^0 * set * admin^0 * sql)
 
 return M
+
```

Comment:
The extra blank line can be remove, the other reformatting is fine.  The grammar looks reasonable (without looking at the Percona source) but I am betting Schema can also be an empty string in their output too.